### PR TITLE
PP-844: Make resource_release_list and resources_released attribute visible to operator/manager only

### PR DESCRIPTION
**Pull request by arungrover**
…ys it should be

#### Issue-ID

* **[PP-844](https://pbspro.atlassian.net/browse/PP-844)**

#### Problem description
resources_released_list attribute isn't visible to the user and design say it should be. 

#### Cause / Analysis
Actually none of the resource_released attributes needs to be visible to normal user. since the attribute which specifies what all resources to be released on suspension is settable only by admins, it does not help users to view what all were actually released.

#### Solution description
Made the resources_released and resource_released_list viewable only to operator/manager privilege user.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [x] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [x] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by arungrover** [comment]
This pull resulted into changes in the design document present here - https://pbspro.atlassian.net/wiki/spaces/PD/pages/51380964/PP-734+Ability+to+free+limited+resources+when+a+job+is+suspended

File: test/tests/functional/pbs_release_limited_res_suspend.py [hunk 1]
```diff
@@ -756,3 +756,28 @@ def test_resuming_excljob_with_no_res_released(self):
         # resume job
         self.server.deljob(jid2, wait=True)
         self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+    def test_normal_user_unable_to_see_res_released(self):
+        """
+        Check if normal user (non-operator, non-manager) has priveleges to see
```
Comment (vccardenas) [inline test/tests/functional/pbs_release_limited_res_suspend.py]: priveleges should be privileges.

File: test/tests/functional/pbs_release_limited_res_suspend.py [hunk 2]
```diff
@@ -756,3 +756,28 @@ def test_resuming_excljob_with_no_res_released(self):
...
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        # suspend job
+        self.server.sigjob(jobid=jid1, signal="suspend")
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+
+        # stat the job as a normal user
+        attrs = self.server.status(JOB, id=jid1, runas=TEST_USER)
+        self.assertFalse("resources_released" in attrs,
+                         "normal user can not see read resources_released")
+
```
Comment (vccardenas) [inline test/tests/functional/pbs_release_limited_res_suspend.py]: Please get rid of "read" to make this failure message similar to that below for resource_released_list.mem.

File: test/tests/functional/pbs_release_limited_res_suspend.py [hunk 3]
```diff
@@ -756,3 +756,28 @@ def test_resuming_excljob_with_no_res_released(self):
...
+        self.server.sigjob(jobid=jid1, signal="suspend")
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+
+        # stat the job as a normal user
+        attrs = self.server.status(JOB, id=jid1, runas=TEST_USER)
+        self.assertFalse("resources_released" in attrs,
+                         "normal user can not see resources_released")
+
+        self.assertFalse("resource_released_list.mem" in attrs,
+                         "normal user can not see resource_released_list")
```
Comment (bhroam) [inline test/tests/functional/pbs_release_limited_res_suspend.py]: I think these messages are backwards.  You're asserting that 'resources_released' is not in attrs.  If the assert fails (so resources_released is in attrs), you say 'normal user can not see resources_released'.  They can see resources_released and that is the problem.  Maybe change the message to 'Normal user can see resources_released which is not expected'

Comment (vccardenas) [reply test/tests/functional/pbs_release_limited_res_suspend.py]: I took "can not see" as "should not see" but the suggestion from @bhroam is clearer.

Comment (arungrover) [reply test/tests/functional/pbs_release_limited_res_suspend.py]: how about "normal user must not see <attr>"?

Comment (vccardenas) [reply test/tests/functional/pbs_release_limited_res_suspend.py]: I suggest "Normal user is not expected to see ..."

Comment (lisa-altair) [reply test/tests/functional/pbs_release_limited_res_suspend.py]: I like @bhroam suggestion best.

**Comment by arungrover** [comment]
I've made changes as per Bhroam's suggestion. Please have a look again

**Comment by vccardenas** [comment]
The changes look good to me - I sign off.

**Comment by bhroam** [comment]
Please rebase and squash for the merge.

**Comment by arungrover** [comment]
sorry about the delay, I've pushed the rebased version.

**Comment by bhroam** [comment]
I merged in the PR.

**Pull request merged**